index.html

```diff
@@ -91,43 +95,65 @@ <h2>
         Badge model
       </h2>
       <p>
-        There is a single global badge associated with each <a data-cite=
-        "appmanifest#installable-web-applications">web application</a>. At any time, the application's badge
-        is set to one of the following values:
+        A <dfn>badge</dfn> may be one of the following values:
       </p>
       <ul>
-        <li>The special value <a>nothing</a>.
+        <li>The special value <dfn>nothing</dfn>.
         </li>
-        <li>The special value <a>flag</a>.
+        <li>The special value <dfn>flag</dfn>.
         </li>
         <li>An <code><a data-cite="WEBIDL#idl-unsigned-long-long">unsigned long
         long</a></code> value, which MUST NOT be 0.
         </li>
       </ul>
       <p>
-        The user agent MUST initialize each application's badge to
-        <a>nothing</a>. The user agent MAY reset an application's badge to
-        <a>nothing</a> at its discretion (for example, when the system is
-        restarted).
+        Each <a data-cite="dom#concept-document">document</a> and each
+        <a data-cite="appmanifest#installable-web-applications">web
+        application</a> is associated with a <a>badge</a> value, which is
+        initialized to <a>nothing</a>.
+      </p>
+      <p>
+        The user agent MAY reset an application's badge to <a>nothing</a> at
+        its discretion (for example, when the system is restarted).
       </p>
       <p>
-        If the application's badge is <dfn>nothing</dfn>, the badge is said to be
+        If a <a>badge</a> is <a>nothing</a>, it is said to be
         "<dfn>cleared</dfn>". Otherwise, it is said to be "<dfn>set</dfn>".
```

Comment:
I don't know what you're asking for here. Do you really want me to switch to set notation here?

This para is where we define "set" and "cleared" terms. I think it's quite understandable to say "If badge is nothing, it is cleared. Otherwise it is set."

Note that this *isn't* an appropriate place to define "nothing" (see above). This is essentially defining some new terms in terms of the already-defined terms.

More Haskell:

```haskell
data SetOrCleared = Set | Cleared
badgeSet Nothing = Cleared
badgeSet Flag = Set
badgeSet (Number _) = Set
```

Could we just rewrite all web specs in Haskell?